### ISSUE TITLE
Tagging feature not available for api gateway stage using SAM model

### ISSUE THREAD

**Issue by arcsoftech** [awslabs/serverless-application-model#384]
It seems that resource tagging for resource type AWS::Serverless::Api is not available  using SAM model.
Also tagging is not working from the global section like we define in cloudformation template.

**Comment by jfuss** [comment]
@arcsoftech At the moment, Tagging for `AWS::Serverless::Api` is not supported. Can you update this request to include how you envision this being added to the `AWS::Serverless::Api`? You can use #376 as an example. This will help in creating discussions on how the schema should look for this feature.

**Comment by arcsoftech** [comment]
@jfuss It could be done same like we do for Lambda function as in "AWS::Serverless::Function" and translated to final CFN.

Here is how I envision it:-
**SAM Template:**
`  ApiGatewayApi:
    Type: AWS::Serverless::Api
    Properties:
      DefinitionUri: ./apigateway-proxy-api.yaml
      StageName: !Ref AppEnv
      Variables:
        lbfunc: !Ref medibotApp
      Tags:
        parent_project: !Ref ParentProjectTag
        environment: !Ref EnvironmentTag
        billing_ref: !Ref BillingRefTag
        application: !Ref ApplicationTag
        key_contact: !Ref KeyContact
        project: !Ref ProjectTag
        STAGE: !Ref STAGETag`
**Transformed CFN Template:**
`Type: "AWS::ApiGateway::Stage"
Properties:

  MethodSettings:
    - MethodSetting
  StageName: String
  Variables:
    String: String
Tags:
   - Key: parent_project
     Value:
       Ref: ParentProjectTag
   - Key: environment
     Value:
       Ref: EnvironmentTag
   - Key: billing_ref
     Value:
       Ref: BillingRefTag
   - Key: application
     Value:
       Ref: ApplicationTag
   - Key: key_contact
     Value:
       Ref: KeyContact
   - Key: project
     Value:
       Ref: ProjectTag
   - Key: STAGE
     Value:
       Ref: STAGETag`

**Comment by arcsoftech** [comment]
Any updates on the above ?

**Comment by Nr18** [comment]
I saw that they used this issue as an example on a twitch episode https://clips.twitch.tv/PoorAthleticWalrusPeteZarollTie 

They implemented in that episode but it looks like it has not been committed yet.

**Comment by ramanganapathy** [comment]
Need this feature for our application development too

**Comment by arcsoftech** [comment]
@Nr18 I sm not able to view that link.Can you explain what they did?

**Comment by Nr18** [comment]
They basically implemented this feature as an example full episode: https://www.twitch.tv/videos/256907743?collection=rjHIYEY1KRXuKA

**Comment by arcsoftech** [comment]
@sanathkr May be I figured out why you got the invalid document exception in the above video/episode .In the transformed template Tags is not inside the "Properties" key but instead it is outside .Can I make those changes and make pr for the same .Just verifying if you already figured it out and merged the changes.Thanks

**Comment by baynezy** [comment]
Is anyone actively working on this? If not I may attempt it myself.

**Comment by goldpenguin93** [comment]
I am currenttly looking for severless study source.

2018년 8월 28일 (화) 오후 9:20, Simon Baynes <notifications@github.com>님이 작성:

**Comment by baynezy** [comment]
@goldpenguin93
Are you close to raising a PR?

**Comment by goldpenguin93** [comment]
what is mean? that PR close?
I m studing Sverless Compute.

2018년 8월 28일 (화) 오후 9:46, Simon Baynes <notifications@github.com>님이 작성:

**Comment by baynezy** [comment]
@goldpenguin93 - I meant do you have a [pull request](https://help.github.com/articles/about-pull-requests/) that you are nearly ready to create?

**Comment by goldpenguin93** [comment]
What is ready?
I went to your question because I am not understnad Alright?


2018년 9월 2일 (일) 오전 12:25, James Hood <notifications@github.com>님이 작성:

**Comment by goldpenguin93** [comment]
2018년 9월 2일 (일) 오전 12:29, Sanghyup <sanghyup1993@gmail.com>님이 작성:

**Comment by jlhood** [comment]
@goldpenguin93 - I think @baynezy was thinking about submitting a fix for this issue, but was trying to determine if you were already close to submitting a fix for it, in which case they would not submit a fix since you were already working on it.

Submitting a fix on a GitHub repo is initiated by created a Pull Request (PR), which is why they asked if you were close to "raising a PR" (creating a PR). You can read more about Pull Requests [here](https://help.github.com/articles/about-pull-requests/).

I was advising @baynezy to go ahead and create their own PR to fix this problem since it's unclear at this point if you are going to create one.

**Comment by goldpenguin93** [comment]
I am understand PR (self introduce) but Now, I am PR(Pull Request)
Understand. Also, I am coming soon PR(pull request) create.

2018년 9월 2일 (일) 오전 12:37, James Hood <notifications@github.com>님이 작성:

**Comment by jlhood** [comment]
Great! Looking forward to your contribution!

**Comment by goldpenguin93** [comment]
Yap! I am comfirm

2018년 9월 4일 (화) 오전 3:45, James Hood <notifications@github.com>님이 작성:

**Comment by marty-sullivan** [comment]
Is adding Tags supported on the standard CFN Stage resource? It isn't listed as a property in the docs:

https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-stage.html

**Comment by hhagblom** [comment]
> Is adding Tags supported on the standard CFN Stage resource? It isn't listed as a property in the docs:
> 
> https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-apigateway-stage.html

It seems to be supported according to the link you sent, is it newly added? Anybody working on a PR or what is the status?

**Comment by marty-sullivan** [comment]
@hhagblom Yes, that does seem to be a recent add, Tags were definitely not there before. In that case, tags should also be added to the SAM resource.

**Comment by keetonian** [comment]
The docs do say that tags are now supported. 

Tags should be added as a parameter to the ApiGatewayStage class: https://github.com/awslabs/serverless-application-model/blob/master/samtranslator/model/apigateway.py#L28

Tags should be added as a parameter to the SamApi class: https://github.com/awslabs/serverless-application-model/blob/master/samtranslator/model/sam_resources.py#L428

Add the tags in the api generator: in the [init](https://github.com/awslabs/serverless-application-model/blob/master/samtranslator/model/api/api_generator.py#L26) statement as well as when the [Stage is being created](https://github.com/awslabs/serverless-application-model/blob/master/samtranslator/model/api/api_generator.py#L151)

**Comment by eduardovra** [comment]
I created a PR to add this feature (#993).

**Comment by goldpenguin93** [comment]
I wonder to PR? What is mean?

2019년 6월 23일 (일) 오전 4:41, Eduardo Vieira <notifications@github.com>님이 작성:

**Comment by Nr18** [comment]
@goldpenguin93 it means pull request

**Comment by goldpenguin93** [comment]
Ok, I undersand PR
Also, soon  I hands on to Pull Request.

2019년 6월 23일 (일) 오후 8:05, Joris Conijn <notifications@github.com>님이 작성:

**Comment by jlhood** [comment]
Pending v1.14.0 release

**Comment by rynop** [comment]
Is it also not possible to add `Tags` to `AWS::Serverless::Function`? I get an error when trying to do so.

**Comment by jlhood** [comment]
@rynop `AWS::Serverless::Function` supports tags ([spec reference](https://github.com/awslabs/serverless-application-model/blob/master/versions/2016-10-31.md#awsserverlessfunction)), but note, the syntax is slightly different if you're used to the native CloudFormation `AWS::Lambda::Function` resource.

For `AWS::Lambda::Function`, it looks like this:

```yaml
MyFunction:
  Type: AWS::Lambda::Function
  Properties:
    Tags:
      - Key: Foo
        Value: Bar
      - Key: Baz
        Value: Blah
    ...
```

For `AWS::Serverless::Function`, the equivalent looks like this:

```yaml
MyFunction:
  Type: AWS::Serverless::Function
  Properties:
    Tags:
      Foo: Bar
      Baz: Blah
    ...
```

**Comment by ShreyaGangishetty** [comment]
Closing this issue as this is released in v1.14.0
